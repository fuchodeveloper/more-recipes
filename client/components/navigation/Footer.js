import React from 'react';
import { render } from 'react-dom';

class Footer extends React.Component {
  render() {
    return (
      <footer>
        <p className="text-center">
            <strong>&copy; 2017 | More-recipes</strong>
        </p>
      </footer>
    )
  }
}

export default Footer;
