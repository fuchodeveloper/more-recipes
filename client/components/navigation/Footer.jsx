import React from 'react';

const Footer = () => (
  <footer>
    <p className="text-center">
      <strong>&copy; {new Date().getFullYear()} | More-recipes</strong>
    </p>
  </footer>
);

export default Footer;
