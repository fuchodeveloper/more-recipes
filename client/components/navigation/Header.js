import React from 'react';
import { Link } from 'react-router-dom';
import { render } from 'react-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logout } from '../../action/authentication/loginAction';

class Header extends React.Component {
  logout(e) {
    e.preventDefault();
    this.props.logout();
  }

  /**
   * 
   * 
   * @returns 
   * @memberof Header
   */
  render() {

    const { isAuthenticated } = this.props.login;
    /**
     * Links available to logged in users
     */

    const userLinks = (
     <div>
       <div className="dropdown show mr-5 main-nav-menu">
          <a className="dropdown-toggle btn btn-block" href="#" role="button" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
              John
          </a>

          <div className="dropdown-menu main-nav-menu width-5" aria-labelledby="dropdownMenuLink">
            <a className="dropdown-item" href="template/profile.html">Profile</a>
            <Link className="dropdown-item" to="add_recipe">Add Recipe</Link>
            <a className="dropdown-item" href="template/category.html">Category</a>
            <a className="dropdown-item" href="template/favorite_recipes.html">Favorite Recipes</a>
            <a className="dropdown-item" href="template/my_recipes.html">My Recipes</a>
            <a className="dropdown-item" href="#" onClick={this.logout.bind(this)}>Logout</a>
          </div>
        </div>
     </div>
    );

    /**
     * Links available to guests
     */
    const guestLinks = (
      <div>
        <Link to="/signup" className="mr-2 btn">Sign Up</Link>
        <Link to="/login" className="mr-2 btn">Sign In</Link>
      </div>
    );

    return (
      <div>
        <nav className="navbar navbar-expand-md navbar-dark navbar-bg-color fixed-top">
          <a className="navbar-brand navbar-brand-color" href="/">More Recipes</a>
          <button className="navbar-toggler hamburger-color-primary" type="button" data-toggle="collapse" data-target="#navbarsExampleDefault" aria-controls="navbarsExampleDefault" aria-expanded="false" aria-label="Toggle navigation">
              <span className="navbar-toggler-icon"/>
          </button>

          <div className="collapse navbar-collapse" id="navbarsExampleDefault">
              <nav className="navbar-nav ml-auto main-nav-menu">
                  <Link to="/" className="mr-2 btn nav-menu-active">Home</Link>
                  
                  {/* <div className="dropdown-divider"/>
                  <Link to="/signup" className="mr-2 btn">Sign Up</Link>
                  <div className="dropdown-divider"/>
                  <Link to="/login" className="mr-2 btn">Sign In</Link>
                  <div className="dropdown-divider"/> */}


                  {/* <div className="dropdown show mr-5 main-nav-menu">
                      <a className="dropdown-toggle btn btn-block" href="#" role="button" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                          John
                      </a>

                      <div className="dropdown-menu main-nav-menu width-5" aria-labelledby="dropdownMenuLink">
                          <a className="dropdown-item" href="template/profile.html">Profile</a>
                          <a className="dropdown-item" href="template/add_recipe.html">Add Recipe</a>
                          <a className="dropdown-item" href="template/category.html">Category</a>
                          <a className="dropdown-item" href="template/favorite_recipes.html">Favorite Recipes</a>
                          <a className="dropdown-item" href="template/my_recipes.html">My Recipes</a>
                      </div>
                  </div> */}
                  { isAuthenticated ? userLinks : guestLinks }
              </nav>
          </div>
        </nav>
      </div>
    )
  }
}

Header.propTypes = {
  login: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired
}
  
function mapStateToProps(state) {
  return {
      login: state.login
  };
}
  

export default connect(mapStateToProps, { logout })(Header);