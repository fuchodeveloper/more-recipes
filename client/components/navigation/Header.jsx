import React from 'react';
import { Link } from 'react-router-dom';
import { render } from 'react-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import capitalize from 'lodash.capitalize';
import { logout } from '../../action/authentication/loginAction';

class Header extends React.Component {
  logout(e) {
    e.preventDefault();
    this.props.logout();
  }

  /**
   *
   *
   * @returns
   * @memberof Header
   */
  render() {
    const { isAuthenticated, user: { firstName } } = this.props.login;
    /**
     * Links available to logged in users
     */

    const userLinks = (
      <div className="navbar-nav ml-auto main-nav-menu">
        <div className="dropdown show mr-5 main-nav-menu">
          <a className="dropdown-toggle btn" href="#" role="button" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
            <strong>{capitalize(this.props.login.user.firstName)}</strong>
          </a>

          <div className="dropdown-menu main-nav-menu width-5" aria-labelledby="dropdownMenuLink">
            <Link className="dropdown-item" to="/profile">Profile</Link>
            <Link className="dropdown-item" to="/add">Add Recipe</Link>
            <Link className="dropdown-item" to="/categories">Categories</Link>
            <Link className="dropdown-item" to="/favorites">Favorite Recipes</Link>
            <Link className="dropdown-item" to={`/${firstName}/recipes`}>My Recipes</Link>
            <Link className="dropdown-item" to="/votes">Most Upvoted</Link>
            <a className="dropdown-item" href="#" onClick={this.logout.bind(this)}>Logout</a>
          </div>
        </div>
      </div>
    );

    /**
     * Links available to guests
     */
    const guestLinks = (
      <div className="navbar-nav ml-auto main-nav-menu">
        <Link className="nav-item nav-link" to="/signup">Sign Up</Link>
        <Link className="nav-item nav-link" to="/login">Sign In</Link>
      </div>
    );

    return (
      <div>
        <nav className="navbar navbar-expand-lg navbar-light bg-light fixed-top">
          <a className="navbar-brand navbar-brand-color"
           href="/">More Recipes</a>
          <button 
            className="navbar-toggler hamburger-color-primary"
            type="button" 
            data-toggle="collapse" 
            data-target="#navbarNavAltMarkup" 
            aria-controls="navbarNavAltMarkup" 
            aria-expanded="false" 
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon text-white" />
          </button>
          <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
            { isAuthenticated ? userLinks : guestLinks }
          </div>
        </nav>
      </div>
    );
  }
}

Header.propTypes = {
  login: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    login: state.login
  };
}


export default connect(mapStateToProps, { logout })(Header);
