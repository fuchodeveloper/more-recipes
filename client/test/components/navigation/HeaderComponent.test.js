/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import { Header } from '../../../components/navigation/Header';

configure({ adapter: new Adapter() });

/**
 * @description function to setup props used in component
 * @returns {Object} props, enzymeWrapper
 */
const setup = () => {
  const props = {
    logoutProps: jest.fn(),
    auth: {
      user: { firstName: '' }
    }
  };
  const enzymeWrapper = shallow(<Header {...props} />);

  return {
    props,
    enzymeWrapper
  };
};

describe('<Header />', () => {
  it('should check for the site brand name', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('a').text()).toBe('More Recipes');
    expect(enzymeWrapper.find('nav').hasClass('navbar')).toBe(true);
  });
});

