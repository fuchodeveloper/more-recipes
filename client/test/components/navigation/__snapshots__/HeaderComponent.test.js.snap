// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Header component should render with right amount of elements 1`] = `undefined`;
